backend/api/services.py

```diff
@@ -215,3 +221,172 @@ def get_reddit_posts(subreddit, symb, time="week") -> pd.DataFrame:
+    )
+
+    # publicity score by summing pub metrics
+    tweets_df["pub_score"] = tweets_df[
+        ["retweet_count", "reply_count", "like_count", "quote_count"]
+    ].sum(axis=1)
+    tweets_df["sentiment"] = tweets_df["text"].apply(get_sentiment)
+
+    # sort by publicity
+    tweets_df.sort_values(by="pub_score", ascending=False, inplace=True)
+    return tweets_df
+
+
+# NEWS
+# functions from https://github.com/iAhsanJaved/FetchGoogleNews
+def get_news(search_term, data_filter=None) -> pd.DataFrame:
+    """Search through Google News with the "search_term" and get the headlines
+    and the contents of the news that was released today, this week, this month,
+    or this year ("date_filter").
+
+    Args:
+        search_term (str): Symbol/search query to search
+        data_filter (str, optional): Date window - "today", "this_week", "this_month", "this_year". Defaults to None.
+
+    Returns:
+        pd.DataFrame: DataFrame of news and other relevant information.
+    """
+
+    def get_text(x):
+        start = x.find("<p>") + 3
+        end = x.find("</p>")
+        return x[start:end]
+
+    url = clean_url(search_term, data_filter)
+    response = requests.get(url)
+    # get the root directly as we have text file of string now
+    root = ET.fromstring(response.text)
+    # get the required data
+    title = [i.text for i in root.findall(".//channel/item/title")]
+    link = [i.text for i in root.findall(".//channel/item/link")]
+    description = [i.text for i in root.findall(".//channel/item/description")]
+    pubDate = [i.text for i in root.findall(".//channel/item/pubDate")]
+    source = [i.text for i in root.findall(".//channel/item/source")]
+    # clear the description
+    short_description = list(map(get_text, description))
+
+    # set the data frame
+    df = pd.DataFrame(
+        {
+            "title": title,
+            "link": link,
+            "description": short_description,
+            "date": pubDate,
+            "source": source,
+        }
+    )
+    # adjust the date column
+    df.date = df.date.astype("datetime64")
+    df.sentiment = df.title.apply(get_sentiment)
+    return df
+
+
+def clean_url(searched_item, data_filter):
+    """
+    OUTPUT : url to be fecthed for the searched_item and data_filter
+     ---------------------------------------------------
+    Parameters:
+      today' - get headlines of the news that are released only in today
+                       'this_week' - get headlines of the news that are released in this week
+                       'this month' - news released in this month
+                       'this_year' - news released in this year
+                        number : int/str input for number of days ago
+                        or '' blank to get all data
+    """
+    x = datetime.today()
+    today = str(x)[:10]
+    yesterday = str(x + pd.Timedelta(days=-1))[:10]
+    this_week = str(x + pd.Timedelta(days=-7))[:10]
```

Comment:
I think this is a bit of a messy implementation. Native datetime supports date math with the `timedelta` function and you can extract the date component with `.date()` and the text version with `.isoformat()`
![image](https://user-images.githubusercontent.com/13448687/202839540-776dce57-39ed-4033-8faa-b56569ce631b.png)